lib/pure/rationals.nim

```diff
@@ -40,17 +40,17 @@ func reduce*[T: SomeInteger](x: var Rational[T]) =
     reduce(r)
     doAssert r.num == 1
     doAssert r.den == 2
-
+  if x.den == 0:
+    raise newException(DivByZeroDefect, "division by zero")
   let common = gcd(x.num, x.den)
   if x.den > 0:
     x.num = x.num div common
     x.den = x.den div common
-  elif x.den < 0:
-    when T isnot SomeUnsignedInt:
-      x.num = -x.num div common
-      x.den = -x.den div common
-  else:
-    raise newException(DivByZeroDefect, "division by zero")
+    return
```

Comment:
I didn't ask for this `return`.

Reply:
sorry, i dont know what you want instead.

would you prefer
```nim
if x.den==0: raise DivByZeroDefect
if x.den > 0:
  blah
when T isnot SomeUnsignedInt:
  if x.den < 0:
    blah
 ```
 ?